openedx/core/djangoapps/oauth_dispatch/migrations/0003_application_data.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.13 on 2018-06-05 13:19
+from __future__ import unicode_literals
+
+from django.conf import settings
+from django.db import migrations
+
+
+def migrate_application_data(apps, schema_editor):
+    """
+    Migrate existing DOT Application models to new ScopedApplication models.
+    """
+    Application = apps.get_model(settings.OAUTH2_PROVIDER_APPLICATION_MODEL)
+    ScopedApplication = apps.get_model('oauth_dispatch', 'ScopedApplication')
+
+    for application in Application.objects.all():
+        ScopedApplication.objects.get_or_create(
+            id=application.id,
+            defaults={
+                'client_id': application.client_id,
+                'user': application.user,
+                'redirect_uris': application.redirect_uris,
+                'client_type': application.client_type,
+                'authorization_grant_type': application.authorization_grant_type,
+                'client_secret': application.client_secret,
+                'name': application.name,
+                'skip_authorization': application.skip_authorization,
+            }
+        )
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('oauth_dispatch', '0002_scopedapplication_scopedapplicationorganization'),
+    ]
+
+    operations = [
+        migrations.RunPython(migrate_application_data, reverse_code=migrations.RunPython.noop),
```

Comment:
I think you may actually want to implement a proper reverse here because if we end up making changes to the new table, we will probably want to propagate them back to the old table so auth does not break.  This is an unlikely case but I'd rather have it implemented correctly if we need it.

Reply:
We won't need it. The new table will not be accessible from the app until we are certain we won't be rolling back.